src/codegen/compile_enhanced.ml

```diff
@@ -6863,7 +6899,10 @@ module Serialization = struct
           add_idx f.typ
         ) fs
       | Mut t ->
-        add_sleb128 idl_alias; add_idx t
+         add_sleb128 idl_alias; add_idx t
+      | Weak t -> (* TODO  - probably want another idl_type, idl_weak, for Weak, otherwise we can
+         deserialize one as the other *)
+         add_sleb128 idl_alias; add_idx (Weak t)
```

Comment:
Do we need to serialize the weak ref? It is not used in cross-canister messages and not in classical persistence IIUC

Reply:
Probably not. I was working under the assumption that we would add them to the classic compiler too, eventually, in which case we would need destabilization logic. But safe to remove for now.